src/lib.rs

```diff
@@ -973,3 +980,29 @@ fn getenv_unwrap(v: &str) -> String {
 fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
+
+// Returns the major, minor and patch versions of cmake
+fn cmake_version(executable: &OsStr) -> Result<(u8, u8, u8), Box<dyn std::error::Error>> {
```

Comment:
I don't think this error is ever read, so it's fine to return an `Option<T>` from this function.